pkg/utils/upload/upload.go

```diff
@@ -0,0 +1,41 @@
+package uploadutils
+
+import (
+	"context"
+	"crypto/md5"
+	"encoding/hex"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"cloud.google.com/go/storage"
+)
+
+// MD5FromGCS returns the MD5 encoding from the GCS object's metadata.
+func MD5FromGCS(ctx context.Context, hd *storage.BucketHandle, path string) (string, error) {
+	attrs, err := hd.Object(path).Attrs(ctx)
+	if err != nil {
+		return "", fmt.Errorf("failed to get attrs for obj: %v", err)
+	}
+	return hex.EncodeToString(attrs.MD5), nil
```

Comment:
this does basically the same as fmt.Sprintf("%x", )
right? why not do the same thing (this hex.EncodeToString() or Sprintf) for this checksum return value? (in MD5FromHTTP)

Reply:
Oh, I copied this from the mass uploader - thought this is the right way to convert MD5.